### DISCUSSION TITLE
"Error in routine phoebe (1):      Cholesky failed" for phoebe-quantum-espresso

### DISCUSSION THREAD

**Discussion by TiacWei** [phoebe-team/phoebe#224] [Q&A] [answered]
I have successfully tested the Si-epa example, and it works well with both Phoebe-Quantum ESPRESSO and Phoebe.

Subsequently, I applied the workflow to my own structure:
1. SCF: This step completed successfully.
2. Ph: This step failed with the error: `Error in routine phoebe (1): Cholesky failed`.
3. NSCF: This step also failed with the same error: `Error in routine phoebe (1): Cholesky failed`.

Here are the troubleshooting steps I tried:
1. I tested almost all the parameters in the input file, but the process failed for the same reason each time.
2. I tried using a different structure (CdSe), and it worked without issues.
3. I used the same input files with unmodified Quantum ESPRESSO version 7.2, and it also worked fine.

Based on these observations, I suspect that there might be a bug affecting certain structures.

Attached are my input files and the error reports for your reference.

```scf.in
&CONTROL
  prefix = 'Sr3SbN'
  calculation = 'scf'
  pseudo_dir = './'
  outdir = "./out"
  verbosity = "high"
  wf_collect = .true.
/
&SYSTEM
  ibrav= 0,   
  nat = 5
  ntyp = 3 
  ecutwfc = 80
  nbnd = 30
/
&ELECTRONS
   conv_thr = 1.0d-10
   diagonalization = "cg"
/
ATOMIC_SPECIES
N      14.0067 N.oncvpsp.upf
Sb     121.76  sb_pbe_v1.4.uspp.F.UPF
Sr     87.62   Sr_pbe_v1.uspp.F.UPF
CELL_PARAMETERS (angstrom)
   5.206015206   0.000000000   0.000000000
   0.000000000   5.206015206   0.000000000
   0.000000000   0.000000000   5.206015206
ATOMIC_POSITIONS (crystal)
Sr       0.000000000   0.500000000   0.500000000
Sr       0.500000000   0.500000000   0.000000000
Sr       0.500000000  -0.000000000   0.500000000
Sb       0.000000000   0.000000000   0.000000000
N        0.500000000   0.500000000   0.500000000
K_POINTS automatic
7 7 7 0 0 0

```

```bands.in
&CONTROL
   calculation = "bands"
   restart_mode = "from_scratch"
   prefix = "Sr3SbN"
   pseudo_dir = "./"
   outdir = "./out" 
/
&SYSTEM
  ibrav= 0,   
  nat = 5
  ntyp = 3 
  ecutwfc = 80
  nbnd = 32
/
&ELECTRONS
   conv_thr = 1.0d-10
   diagonalization = "cg"
/
ATOMIC_SPECIES
N      14.0067 N.oncvpsp.upf
Sb     121.76  sb_pbe_v1.4.uspp.F.UPF
Sr     87.62   Sr_pbe_v1.uspp.F.UPF
CELL_PARAMETERS (angstrom)
   5.206015206   0.000000000   0.000000000
   0.000000000   5.206015206   0.000000000
   0.000000000   0.000000000   5.206015206
ATOMIC_POSITIONS (crystal)
Sr       0.000000000   0.500000000   0.500000000
Sr       0.500000000   0.500000000   0.000000000
Sr       0.500000000  -0.000000000   0.500000000
Sb       0.000000000   0.000000000   0.000000000
N        0.500000000   0.500000000   0.500000000
K_POINTS crystal
343
  0.00000000  0.00000000  0.00000000  2.915452e-03
  0.00000000  0.00000000  0.14285714  2.915452e-03
......
```

```error
     number of k points=   343

     Number of k-points >= 100: set verbosity='high' to print them.

     Dense  grid:    91965 G-vectors     FFT dimensions: (  60,  60,  60)

     Estimated max dynamical RAM per process >      26.11 MB

     Estimated total dynamical RAM >     835.60 MB

     Check: negative core charge=   -0.000002

     The potential is recalculated from file :
     ./out/Sr3SbN.save/charge-density


     negative rho (up, down):  1.993E-01 0.000E+00
     Starting wfcs are   52 randomized atomic wfcs

     Band Structure Calculation
     CG style diagonalization

 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
     Error in routine phoebe (1):
     Cholesky failed
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

     stopping ...
```

**Comment by jcoulter12** [comment]
Hi @TiacWei, 

Thanks very much for reporting this and also for sharing your input file. 
Would you mind sending also your ph.in file, so that I can reproduce this myself? 

I'll try to look into this as soon as I can. 

Best, 
Jenny

**Reply by TiacWei** [reply]
Hi Jenny

Thank you for your response.
Please find attached all the input files, UPF files, and the run command.
[Sr3SbN.zip](https://github.com/user-attachments/files/16860273/Sr3SbN.zip)


Best,
TiacWei

**Reply by jcoulter12** [reply, marked as answer]
Hi @TiacWei, 

Thanks for your patience and for sending these files. 

I am looking into this a little bit more, but this will work properly for you if you use all ONCV pseudopotentials. Specifically, it seems like there is a subtle point of difficulty writing and reading the gauge of the wavefunction from QE. It seems to be specifically related to the use of your Sr pseudopotential. 

It looks like things should work if you use the ones here: http://www.pseudo-dojo.org/, or at the bare minimum, replace the Sr one with a norm conserving potential. 

Can you confirm that this works also for you? 
Best, 
Jenny

**Reply by TiacWei** [reply]
Hi Jenny,

Thank you so much for your help. Changing the pseudopotentials worked perfectly! Previously, I was using the standard Solid-State Pseudopotentials (SSSP) library, but it seems that the famous SSSP library doesn't always work in every case. Thanks again for your support—now I can proceed with the next steps.

Best regards,
TiacWei
